Parallel decryption in case of many items

peterdeme opened the pull request:
Hello guys,

We have noticed that our app startup is very very slow (~23sec in case of 33 items). The problem is because of credstash config building.
With this change we were able to reduce it to ~5sec.

From our tests it looks like that like 10 is the correct number for degree of parallelism.

Thanks,
Peter

Commit:
Parallel decryption in case of many items

adamhathcock commented:
I'm not 100% sure I like parallel hitting the table as Dynamo can decide to error out if a request limit is hit.

I guess it doesn't hurt to make the degree of parallelism be a config value though.

I see you renamed async functions to be *Async.  I guess I should follow that convention :)

peterdeme committed:
Added DoP as an options, added overloads for configurationbuilder

peterdeme commented:
@adamhathcock you are right. the read unit capacity is important.
- I added it as a configuration option.
- Added a 2 more overloads for IConfigurationBuilder building
- I've added xml description to the dop option, so i've ended up adding explanation for the rest
- Visual Studio kept asking me for fix line endings, so I clicked OK in the popup

 _"you renamed async functions to be *Async"_
Yeah, that's a general C# convention that I like to follow.

Thanks for the review!
Peter
